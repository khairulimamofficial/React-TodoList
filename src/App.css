/* CSS Sidebar */

.sidenav {
  height: 100%;
  width: 250px;
  position: fixed;
  z-index: 1;
  top: 62px;
  left: 0;
  background-color: #F1F1F1;
  overflow-x: hidden;
  padding-top: 20px;
}

.sidenav ul{
  list-style-type: none;
  padding-left: 30px;
}

.sidenav ul li a{
  padding-top: 6px;
  text-decoration: none;
  font-size: 18px;
  color: #707070;
  display: block;
}

.sidenav ul li a:active {
  color: #1B75BC;
  text-decoration: none;
}

.main {
  margin-left: 250px;
  padding: 30px;
}

@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav ul li a {font-size: 18px;}
}

.main-home h1{
  font-size: 24px;
}
body .modal-footer{
  border: none;
}


/* input */
.input-form form{
  margin: 10px 0px;
}
.input-form form input{
  margin-right: 20px;
  border-color: #1B75BC;
}
.input-form form input{
  margin-right: 20px;
  border-color: #1B75BC;
}

::placeholder {
  color: #909090;
  font-size: 14px;
  padding-left: 5px;
  opacity: 1; /* Firefox */
}

:-ms-input-placeholder { /* Internet Explorer 10-11 */
  color: #909090;
  font-size: 14px;
  padding-left: 5px;
}

::-ms-input-placeholder { /* Microsoft Edge */
  color: #909090;
  font-size: 14px;
  padding-left: 5px;
}

.input-form form button{
  font-size: 14px;
  border: none;
  padding: 4px 25px;
  background-color: #1B75BC;
  color: #ffffff;
}


/* Card */

.card .card-body{
  padding: 10px;

}

.button-cust button{
  border: none;
  background-color: transparent;
  padding: 0px;
  margin: 10px;
}

.button-cust2 button{
  border: none;
  background-color: #e1e1e1;
  padding: 5px 10px;
  margin: 10px;
  font-size: 12px;
}
.button-cust2 p{
  padding: 7px 10px;
  font-size: 12px;
  display: inline;
}

/* TodoList progress */
.todoListMain {
  top: 90px;
  left: 700px;
  position: absolute;
}

.card-cust li{
  list-style: none;
  padding: 0px 10px;
}

.todo{
  padding: 0px 10px;
  
}

.theList {
  padding: 0px;
}


/* Router My Profile */
.profile img{
  float: left;
  margin-right: 20px;
}
.profile{
  padding-bottom: 50px;
  background-color: #FFF7C4;
  margin-right: 500px;
}
.new2{
  margin-top: 20px;
  margin-left: 40px;
}
.new2 h1{
  font-size: 24px;
}
.organization{
  margin-bottom: 20px;
}
.organization table td{
  padding: 5px;
  font-size: 18px;   
}
.experiences table td{
  padding: 5px;
  font-size: 18px;
}
