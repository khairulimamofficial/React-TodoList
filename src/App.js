import React, { Fragment } from 'react';
import './App.css';

//logo brand
import logo from './img/Logo-01.png'

// Bootstrap
import { Navbar, Modal } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

// React Router
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import { Home, HowToUse, MyProfile, OurFeature } from './Menu'



//------------ Main --------------------------------------------//



function App() {
  return (
    <Fragment>

       {/*  Navbar bootstrap */}

      <Navbar sticky="top" style={{backgroundColor: "#E1E1E1"}}>
        <Navbar.Brand href="Home">
          <img
            alt="Brand"
            src={logo}
            width="150px"
            className="d-inline-block align-top"
          />
        </Navbar.Brand>
      </Navbar>

      {/* List sidebar dengan router */}

      <Router>
        <div className="sidenav">
          <ul>
            <li><Link to="/">Home</Link></li>
            <li><Link to="/HowToUse">How to use</Link></li>
            <li><Link to="/OurFeature">Our feature</Link></li>
            <li><Link to="/MyProfile">My profile</Link></li>
          </ul>
        </div>

        {/* href="/#" to remove warning karena tag a (tag a untuk link styling)*/}
        
        <Switch>
          <Route exact path="/">
            <Home/>
          </Route>
          <Route path="/HowToUse">
            <HowToUse/>
          </Route>
          <Route path="/OurFeature">
            <OurFeature/>
          </Route>
          <Route path="/MyProfile">
            <MyProfile/>
          </Route>
        </Switch>
      </Router>

      <Modal.Footer>
        <p style={{top:"95%",left:"82%",position: "fixed", color:"#909090"}}>Copyright by Khairul Imam</p>
      </Modal.Footer>
    </Fragment>
  );
}

export default App;