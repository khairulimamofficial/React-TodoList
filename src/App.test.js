import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import ButtonProgress from './App'

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Make your own list/i);
  expect(linkElement).toBeInTheDocument();
});

test('add task snapshot', ()=> {
  const component = renderer.create(
    <App/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})