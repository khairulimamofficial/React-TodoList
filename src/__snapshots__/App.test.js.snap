// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add task 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light sticky-top"
    style={
      Object {
        "backgroundColor": "#E1E1E1",
      }
    }
  >
    <a
      className="navbar-brand"
      href="Home"
    >
      <img
        alt="Brand"
        className="d-inline-block align-top"
        src="Logo-01.png"
        width="150px"
      />
    </a>
  </nav>,
  <div
    className="sidenav"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/HowToUse"
          onClick={[Function]}
        >
          How to use
        </a>
      </li>
      <li>
        <a
          href="/OurFeature"
          onClick={[Function]}
        >
          Our feature
        </a>
      </li>
      <li>
        <a
          href="/MyProfile"
          onClick={[Function]}
        >
          My profile
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="main"
  >
    <div
      className="todo-list"
    >
      <div
        className="input-form"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Enter task"
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            Add
          </button>
        </form>
      </div>
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "",
            "width": "18rem",
          }
        }
      >
        <div
          className="card-cust card-body"
          style={
            Object {
              "color": "",
            }
          }
        >
          <div
            className="todo"
            style={
              Object {
                "color": "",
              }
            }
          >
            Make your own list
            <div
              className="button-cust"
            >
              <button
                onClick={[Function]}
              >
                <img
                  alt="fotoku"
                  height="20px"
                  src="check.png"
                />
              </button>
              <button
                onClick={[Function]}
              >
                <img
                  alt="fotoku"
                  height="20px"
                  src="trash.png"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="TodoListProgress"
    >
      <div
        className="todoListMain input-form"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            placeholder="Enter progress"
          />
          <button
            type="submit"
          >
            Add Progress
          </button>
        </form>
        <ul
          className="theList"
        />
      </div>
    </div>
  </div>,
  <div
    className="modal-footer"
  >
    <p
      style={
        Object {
          "color": "#909090",
          "left": "82%",
          "position": "fixed",
          "top": "95%",
        }
      }
    >
      Copyright by Khairul Imam
    </p>
  </div>,
]
`;

exports[`add task snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light sticky-top"
    style={
      Object {
        "backgroundColor": "#E1E1E1",
      }
    }
  >
    <a
      className="navbar-brand"
      href="Home"
    >
      <img
        alt="Brand"
        className="d-inline-block align-top"
        src="Logo-01.png"
        width="150px"
      />
    </a>
  </nav>,
  <div
    className="sidenav"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/HowToUse"
          onClick={[Function]}
        >
          How to use
        </a>
      </li>
      <li>
        <a
          href="/OurFeature"
          onClick={[Function]}
        >
          Our feature
        </a>
      </li>
      <li>
        <a
          href="/MyProfile"
          onClick={[Function]}
        >
          My profile
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="main"
  >
    <div
      className="todo-list"
    >
      <div
        className="input-form"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            className="input"
            onChange={[Function]}
            placeholder="Enter task"
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            Add
          </button>
        </form>
      </div>
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "",
            "width": "18rem",
          }
        }
      >
        <div
          className="card-cust card-body"
          style={
            Object {
              "color": "",
            }
          }
        >
          <div
            className="todo"
            style={
              Object {
                "color": "",
              }
            }
          >
            Make your own list
            <div
              className="button-cust"
            >
              <button
                onClick={[Function]}
              >
                <img
                  alt="fotoku"
                  height="20px"
                  src="check.png"
                />
              </button>
              <button
                onClick={[Function]}
              >
                <img
                  alt="fotoku"
                  height="20px"
                  src="trash.png"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="TodoListProgress"
    >
      <div
        className="todoListMain input-form"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            placeholder="Enter progress"
          />
          <button
            type="submit"
          >
            Add Progress
          </button>
        </form>
        <ul
          className="theList"
        />
      </div>
    </div>
  </div>,
  <div
    className="modal-footer"
  >
    <p
      style={
        Object {
          "color": "#909090",
          "left": "82%",
          "position": "fixed",
          "top": "95%",
        }
      }
    >
      Copyright by Khairul Imam
    </p>
  </div>,
]
`;
